crates/oxc_linter/src/rules/unicorn/no_instanceof_array.rs

```diff
@@ -0,0 +1,107 @@
+use oxc_ast::ast::Expression;
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_formatter::Gen;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use oxc_syntax::operator::BinaryOperator;
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Use `Array.isArray()` instead of `instanceof Array`.")]
+#[diagnostic(severity(warning))]
+struct NoInstanceofArrayDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoInstanceofArray;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Require `Array.isArray()` instead of `instanceof Array`.
+    ///
+    /// ### Why is this bad?
+    /// The instanceof Array check doesn't work across realms/contexts, for example, frames/windows in browsers or the vm module in Node.js.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// array instanceof Array;
+    /// [1,2,3] instanceof Array;
+    /// ```
+    NoInstanceofArray,
+    correctness
+);
+
+impl Rule for NoInstanceofArray {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::BinaryExpression(expr) = node.kind() else { return };
+        if expr.operator != BinaryOperator::Instanceof {
+            return;
+        }
+
+        match &expr.right {
+            Expression::Identifier(identifier) if identifier.name.as_str().eq("Array") => {
```

Comment:
How come you prefer `eq` instead of just `==`? There's also Expression::is_specific_id you can use.